docs/DeveloperGuide.md

```diff
@@ -1,22 +1,118 @@
+- [Product Scope](https://ay2324s2-cs2113-w13-1.github.io/tp/DeveloperGuide.html#product-scope)
+- [User Stories](https://ay2324s2-cs2113-w13-1.github.io/tp/DeveloperGuide.html#user-stories)
+- [Non-Functional Requirements](https://ay2324s2-cs2113-w13-1.github.io/tp/DeveloperGuide.html#non-functional-requirements)
+- [Glossary](https://ay2324s2-cs2113-w13-1.github.io/tp/DeveloperGuide.html#glossary)
+- [Instructions for Manual Testing](https://ay2324s2-cs2113-w13-1.github.io/tp/DeveloperGuide.html#instructions-for-manual-testing)
 
-## Design & implementation
+## Acknowledgements
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
+## Design & Implementation
+
+Here is a quick overview of the main components and how they interact with each other:
+
+- [Parser](https://ay2324s2-cs2113-w13-1.github.io/tp/DeveloperGuide.html#parser-implementation): Reads input
+- [Ui](https://ay2324s2-cs2113-w13-1.github.io/tp/DeveloperGuide.html#ui-implementation): The UI of the App
+- [Game](https://ay2324s2-cs2113-w13-1.github.io/tp/DeveloperGuide.html#game-implementation): The Interface concerning the games
+- [TicTacToe](https://ay2324s2-cs2113-w13-1.github.io/tp/DeveloperGuide.html#tictactoe-implementation): Game 1
+- [Hangman](https://ay2324s2-cs2113-w13-1.github.io/tp/DeveloperGuide.html#hangman-implementation): Game 2
+
+### <span style="color:grey;">Parser Implementation:</span>
+Parser Class (v1.0) reads user input from the standard input stream, checks the input, 
+tests the validity, and conveys the validity back to the class it is called from.
+![img.png](img.png)
+
+Briefly, the Parser class interacts with the main program by reading the inputs and throwing exceptions
+if the inputs are unexpected. These exceptions are then handled by the Duke class or the 
+TicTacToe class. 
+![img_4.png](img_4.png)
+
+Currently, the Parser class only covers the reading of the chosen game by the user and 
+the moves performed in TicTacToe. A future implementation for v2.0 would integrate methods
+to read and verify user input for the HangMan class as well, such as readLetters(String: input) 
+or readHMMove(String: input).
+![img_1.png](img_1.png)
+
+### <span style="color:grey;">UI Implementation:</span>
+The `UI` consists of the AeroCade `logo`, the `help` menu, the `tutorial`, and sends farewell to the user upon `quit`
+
+The `UI` component,
+
+* reads user commands using `Parser` component.
+* displays various ASCII art and messages through `Render`.
+* creates a `tutorial` to assist user in `TimerTutorial`.
+  * Executes tutorial frames using the TimerTask mechanism.
+  * Provides interactive learning experience for users.
+  * Depends on the Ui component for displaying tutorial frames and managing user interaction.
```

Comment:
Standardise the use of either `UI` or `Ui`
- "The `UI` component," , `line 45`
- "Depends on the Ui component ", `line 52`